File: backend/src/main/java/com/oose/labsafety/education/domain/ResearcherCategory.java
package com.oose.labsafety.education.domain;

import com.oose.labsafety.common.domain.Identifiable;
import jakarta.validation.constraints.NotBlank;

public record ResearcherCategory(
        @NotBlank(message = "연구활동종사자 분류 ID는 필수입니다.") String researcherCategoryId,
        @NotBlank(message = "분류명은 필수입니다.") String categoryName,
        String description,
        boolean active
) implements Identifiable {
    @Override
    public String id() {
        return researcherCategoryId;
    }
}

File: src/features/education/domain/ResearcherCategory.js
import { createEntity } from '../../../shared/domain/createEntity';

const schema = {
  researcherCategoryId: { label: '연구활동종사자 분류 ID', required: true },
  categoryName: { label: '분류명', required: true },
  description: { label: '분류 설명' },
  active: { label: '사용 여부', type: 'boolean' },
};

export function createResearcherCategory(input) {
  return createEntity(input, schema);
}
